tidy3d/components/eme/simulation.py

```diff
@@ -887,3 +981,24 @@ def grid(self) -> Grid:
         # This would AutoGrid the in-plane directions of the 2D materials
         # return self._grid_corrections_2dmaterials(grid)
         return grid
+
+    def _monitor_num_transverse_cells(self, monitor: Monitor) -> int:
+        """Total number of cells transverse to propagation axis
+        included in monitor based on simulation grid."""
+
+        def num_transverse_cells_in_monitor(monitor: Monitor) -> int:
+            """Get the number of transverse measurement cells in a
+            monitor given the simulation grid and downsampling."""
+            if not self.intersects(monitor):
+                # Monitor is outside of simulation domain; can happen e.g. for integration surfaces
+                return 0
+            num_cells = self.discretize_monitor(monitor).num_cells
+            # take monitor downsampling into account
+            num_cells = list(monitor.downsampled_num_cells(num_cells))
+            # pop propagation axis
+            num_cells.pop(self.axis)
+            return np.prod(np.array(num_cells, dtype=np.int64))
```

Comment:
maybe
```py
np.prod(np.array(num_cells)).astype(int)
```
unless you want `np.int64` specifically?

Reply:
this was copied from `Simulation`. I'm not sure why int64 was used there, but it is probably safer in case the simulation is very large.

Reply:
Yeah, there was a reason for that. Some machines (I think I've only seen it on Windows) seem to run 32-bit numpy and `np.prod(np.array(num_cells))` can overflow.